lib/api/CInferenceModelDefinition.cc

```diff
@@ -0,0 +1,459 @@
+const std::string JSON_TARGET_TYPE_TAG{"target_type"};
+const std::string JSON_THRESHOLD_TAG{"threshold"};
+const std::string JSON_TRAINED_MODEL_TAG{"trained_model"};
+const std::string JSON_TRAINED_MODELS_TAG{"trained_models"};
+const std::string JSON_TREE_STRUCTURE_TAG{"tree_structure"};
+const std::string JSON_TREE_TAG{"tree"};
+const std::string JSON_WEIGHTED_SUM_TAG{"weighted_sum"};
+const std::string JSON_WEIGHTED_MODE_TAG{"weighted_mode"};
+const std::string JSON_WEIGHTS_TAG{"weights"};
+
+template<typename T>
+void addJsonArray(const std::string& tag,
+                  const std::vector<T>& vector,
+                  rapidjson::Value& parentObject,
+                  TRapidJsonWriter& writer) {
+    rapidjson::Value array = writer.makeArray(vector.size());
+    for (const auto& item : vector) {
+        rapidjson::Value value;
+        value.SetString(core::CStringUtils::typeToString(item), writer.getRawAllocator());
+        array.PushBack(value, writer.getRawAllocator());
+    }
+    writer.addMember(tag, array, parentObject);
+}
+}
+
+void CTree::CTreeNode::addToDocument(rapidjson::Value& parentObject,
+                                     TRapidJsonWriter& writer) const {
+    writer.addMember(JSON_NODE_INDEX_TAG, rapidjson::Value(m_NodeIndex).Move(), parentObject);
+
+    if (m_LeftChild) {
+        // internal node
+        writer.addMember(
+            JSON_SPLIT_FEATURE_TAG,
+            rapidjson::Value(static_cast<std::uint64_t>(m_SplitFeature)).Move(),
+            parentObject);
+        if (m_SplitGain.is_initialized()) {
+            writer.addMember(JSON_SPLIT_GAIN_TAG,
+                             rapidjson::Value(m_SplitGain.get()).Move(), parentObject);
+        }
+        writer.addMember(JSON_THRESHOLD_TAG, rapidjson::Value(m_Threshold).Move(), parentObject);
+        writer.addMember(JSON_DEFAULT_LEFT_TAG,
+                         rapidjson::Value(m_DefaultLeft).Move(), parentObject);
+        switch (m_DecisionType) {
+        case E_LT:
+            writer.addMember(JSON_DECISION_TYPE_TAG, JSON_LT, parentObject);
+            break;
+        }
+        writer.addMember(
+            JSON_LEFT_CHILD_TAG,
+            rapidjson::Value(static_cast<std::uint64_t>(m_LeftChild.get())).Move(),
+            parentObject);
+        writer.addMember(
+            JSON_RIGHT_CHILD_TAG,
+            rapidjson::Value(static_cast<std::uint64_t>(m_RightChild.get())).Move(),
+            parentObject);
+    } else {
+        // leaf node
+        writer.addMember(JSON_LEAF_VALUE_TAG,
+                         rapidjson::Value(m_LeafValue).Move(), parentObject);
+    }
+}
+
+CTree::CTreeNode::CTreeNode(TNodeIndex nodeIndex,
+                            double threshold,
+                            bool defaultLeft,
+                            double leafValue,
+                            size_t splitFeature,
+                            const CTree::CTreeNode::TOptionalNodeIndex& leftChild,
```

Comment:
```suggestion
                            const TOptionalNodeIndex& leftChild,
```
and similar below...

Reply:
done